apps/music/test/marionette/loading_test.js

```diff
@@ -0,0 +1,78 @@
+/* global require, marionette, setup, suite, test */
+'use strict';
+
+var assert = require('assert');
+var Music = require('./lib/music.js');
+
+marionette('Music files loading', function() {
+  var apps = {};
+
+  var client = marionette.client({
+    prefs: {
+      'device.storage.enabled': true,
+      'device.storage.testing': true,
+      'device.storage.prompt.testing': true
+    },
+
+    settings: {
+      'lockscreen.enabled': false,
+      'ftu.manifestURL': null
+    },
+
+    apps: apps
+  });
+
+  var music;
+
+  setup(function() {
+    music = new Music(client);
+
+  });
+
+  suite('Detection of movies.', function () {
+    setup(function() {
+      client.fileManager.removeAllFiles();
+      client.fileManager.add([
+        {
+          // Regular Ogg Vorbis.
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/a.ogg'
+        },
+        {
+          // AAC file.
+          type: 'music',
+          filePath: 'apps/music/test-data/aac-tags.mp4'
+        },
+        {
+          // MPEG4 video file. Shouldn't load.
+          type: 'music',
+          filePath: 'test_media/Movies/gizmo2.mp4'
+        }
+      ]);
+    });
+
+    test('Check the mp4 video isn\'t loaded', function() {
+
+      music.launch();
+      music.waitForFirstTile();
+      music.waitFinishedScanning();
+
+      music.switchToSongsView();
+
+      music.waitForSongsInList(function(songs) {
+        return songs.length >= 2;
+      });
+
+      var songs = music.songsInList;
+      assert.equal(songs.length, 2, 'Wrong number of songs');
+
+      var num = client.executeScript(function() {
+        var w = window.wrappedJSObject;
+        return w.ListView.dataSource.length;
+      });
+      assert.equal(num, 2, 'Data source have wrong number of songs');
```

Comment:
Just counting the songs here doesn't seem like enough. We could have accepted the movie and rejected the .ogg file, for example.  Why not have this executeScript() call return `w.ListView.dataSource.map((record) => record.name).sort()` so that you get a list of the song files, in alphabetical order. Then you can assert that the expected files were accepted and the movie was rejected.